### DISCUSSION TITLE
Support for DNxHD using ADHRv2 atom

### DISCUSSION THREAD

**Discussion by mwjburton** [videovillage/community#100] [Ideas]
Since mid 2020, Avid changed the way that DNxHD (AVdn) files are written to a MOV container and how they are named. Previously (and still used by many/most 3rd party enc/dec software) DNxHD was written with the ACLR (DNx Color Mapping), ARPG and ARES atoms for this codec. The DNxHD codecs were named according to their data rate. e.g. DNx115

Once Avid moved to using the ADHRv2 structure, none of those atoms are now written and all the data is now contained in a single ADHR v2 atom. The codec names are now inline with DNxHR, e.g. DNx115 becomes DNxHD SQ.

DNxHR already used the ADHR v1 atom but also included a separate ACLR extension for Color Mapping. I think this also changed mid 2020 to using the single ADHR v2 atom for all data.

The above preamble was more of a 'this is what I understand' rather than any attempt to educate, you'll be a thousand miles ahead of me on this one.

So, the issue...

Screen v1.7.3
macOS v12.3.1
PVF v2.2.2 and v2.2.3

A DNxHD SQ file exported from Media Composer 2021.12 with limited (tv) colour range reads and displays incorrectly in Screen as RGB when the preference is set to 'Prefer AVFoundation before FFmpeg'. When this preference is disabled, the file reads and displays correctly as YUV.

Perhaps this is related to the fact AVFoundation setting is looking for the ACLR atom and finding none, and assuming incorrectly the file is full range? Although I'm not seeing the same issue with the colour range and this preference with newer DNxHR (AVdh) files that use the ADHR v2 atom and therefore don't have an ACLR atom either.

Note: Older ACLR/ARPG/ARES formatted DNxHD files work as expected.

**Comment by mwjburton** [comment]
If there's anything more I can do to help establish whats happening here, let me know.

**Comment by gregcotten** [comment]
Hey! I think I just need a test file from you - could you send to support@videovillage.co?

**Comment by mwjburton** [comment]
Sent

**Comment by mwjburton** [comment]
Any luck with this Greg?

**Comment by gregcotten** [comment]
@mwjburton Just to make sure - QuickTime also has this issue, yes? I might put a support ticket out with Apple for this.

**Comment by mwjburton** [comment]
@gregcotten Quite possibly thats the case, but I don't see a way to display Colour Space in Quicktime Player. The inspector doesn't show this. Is there another way to tell?

In Screen its much easier to see as your info window is far more useful:

With 'Prefer AVFoundatition before FFmpeg' ticked:       8-bit RGB
With 'Prefer AVFoundatition before FFmpeg' unticked:   8-bit Y'CbCr 4:2:2

**Comment by mwjburton** [comment]
Issue still present after installing latest Pro Video Formats 2.2.5

**Comment by mwjburton** [comment]
Hi Greg, I appreciate that its not going to be a priority, but this bug has been around for 1.5 years now and is pretty serious.

An Avid DNxHD MOV file is still being seen as RGB by Screen when the "Prefer AVFoundation before FFmpeg" preference is enabled. The image is being displayed incorectly and reported as RGB in 'Basic Info'.

The issue does not occur with the same file when the "Prefer AVFoundation before FFmpeg" preference is disabled. However this results in poor performance / scrubbing and isn't usable.

The issue does not occur with Avid DNxHD MXF files.

The issue does not occur with Quicktime Player 10.5.

This is with the latest Media Composer (but has affected every version since first reported), latest Pro Video Formats 2.2.7 (but has affected every version since first reported). MacOS 13.6.1 (but has effected every version since first reported).

It would be great of this could be looked into as its a fairly major bug that a standard Avid file is being read and displayed incorrectly by Screen.

Many thanks
Mark

**Reply by mwjburton** [reply]
Just to be 100% clear, I'm saying the image is being displayed incorrectly as well as the metadata showing incorrectly. This is an image display bug first and foremost.
